#!/usr/bin/env python
PACKAGE = "range_sensor_layer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('enabled',               bool_t, 0, 'Whether to apply this plugin or not', True)
gen.add('phi',                 double_t, 0, 'Phi value', 1.2)
gen.add('inflate_cone',        double_t, 0, 'Inflate the triangular area covered by the sensor (percentage)', 1, 0, 1)
gen.add('no_readings_timeout', double_t, 0, 'No Readings Timeout', 0.0, 0.0)
gen.add('clear_threshold',     double_t, 0, 'Probability below which cells are marked as free', 0.2, 0.0, 1.0)
gen.add('mark_threshold',      double_t, 0, 'Probability above which cells are marked as occupied', 0.8, 0.0, 1.0)
gen.add('clear_on_max_reading',  bool_t, 0, 'Clear on max reading', False)

exit(gen.generate(PACKAGE, PACKAGE, "RangeSensorLayer"))
